core/slice.go

```diff
@@ -217,14 +215,31 @@ func (sl *Slice) Append(header *types.Header, domPendingHeader *types.Header, do
 
 	// Append has succeeded write the batch
 	if err := batch.Write(); err != nil {
-		return nil, err
+		return nil, false, err
 	}
 	appendFinished := time.Since(start)
 	time11 := common.PrettyDuration(appendFinished)
-	reorg := sl.writeToPhCacheAndPickPhHead(pendingHeaderWithTermini, &appendFinished)
+	bestPh, exist := sl.readPhCache(sl.bestPhKey)
+	if !exist {
+		log.Error("BestPh Key does not exist for", "key", sl.bestPhKey)
+	}
+	oldBestPhEntropy := new(big.Int).Set(bestPh.Header.CalcPhS())
+
+	sl.updatePhCache(pendingHeaderWithTermini, true)
+
+	if nodeCtx == common.ZONE_CTX {
+		subReorg = sl.pickPhHead(pendingHeaderWithTermini, oldBestPhEntropy)
+	}
+	if subReorg {
+		if &appendFinished != nil { // If appendTime is nil, it means this was called from updatePhCacheFromDom
```

Comment:
What is going on here? You're taking a reference to a variable, and then checking if that reference is nil? It will never be nil. Am I misunderstanding?

Reply:
removed check